src/backend/columnar/columnar_tableam.c

```diff
@@ -2126,9 +2128,87 @@ ColumnarProcessUtility(PlannedStmt *pstmt,
 
 		RelationClose(rel);
 	}
+	else if (IsA(parsetree, CreateStmt))
```

Comment:
I think now would be a great moment to change the if/elseif pairs here to a switch statement.

Reply:
Done.